presets/ragengine/vector_store/base.py

```diff
@@ -208,6 +206,69 @@ async def add_document_to_index(self, index_name: str, document: Document, doc_i
     def list_indexes(self) -> List[str]:
         return list(self.index_map.keys())
 
+    async def delete_documents(self, index_name: str, doc_ids: List[str]):
+        """Common logic for deleting a document."""
+        if index_name not in self.index_map:
+            raise ValueError(f"No such index: '{index_name}' exists.")
+        
+        not_found_docs = []
+        found_docs = []
+        for doc_id in doc_ids:
+            if doc_id in self.index_map[index_name].ref_doc_info:
+                found_docs.append(doc_id)
+            else:
+                not_found_docs.append(doc_id)
+
+        try:
+            if self.use_rwlock:
+                async with self.rwlock.writer_lock:
+                    await asyncio.gather(*(self.index_map[index_name].adelete_ref_doc(doc_id, delete_from_docstore=True) for doc_id in doc_ids))
```

Comment:
checked through other vector_store implementations in llama_index and most dont implement `delete_nodes` so went for `adelete_ref_doc` so later options will have support out of the gate